ui/pages/confirmations/hooks/useConfirmationNetworkInfo.ts

```diff
@@ -29,11 +29,16 @@ function useConfirmationNetworkInfo() {
     const currentChainId =
       currentConfirmation?.chainId ?? providerConfig.chainId;
     confirmationNetwork = allNetworks.find(
-      ({ chainId }) => chainId === currentChainId,
+      ({ id, chainId }) =>
+        chainId === currentChainId &&
+        (providerConfig.type === 'rpc'
+          ? id === providerConfig.id
+          : id === providerConfig.type),
     );
+
     if (confirmationNetwork) {
-      const { nickname, providerType, type } = confirmationNetwork;
-      if (providerType === NETWORK_TYPES.RPC || type === NETWORK_TYPES.RPC) {
+      const { nickname } = confirmationNetwork;
+      if (providerConfig.type === NETWORK_TYPES.RPC) {
         networkDisplayName = nickname ?? t('privateNetwork');
```

Comment:
[double-checking] if there wasn't confirmationNetwork found and if `providerConfig.type === NETWORK_TYPES.RPC)`, would we want to set networkDisplayName to `t('privateNetwork')`

Reply:
Yep, `providerConfig.type === NETWORK_TYPES.RPC)` is true for custom networks.